nxc/protocols/ssh.py

```diff
@@ -111,76 +209,82 @@ def plaintext_login(self, username, password, private_key=None):
+            self.conn.close()
+            return False
+        else:
             shell_access = False
             host_id = self.db.get_hosts(self.host)[0].id
 
-            if self.check_if_admin():
-                shell_access = True
-                self.logger.debug(f"User {username} logged in successfully and is root!")
-                if self.args.key_file:
-                    self.db.add_admin_user("key", username, password, host_id=host_id, cred_id=cred_id)
+            if not stdout:
+                stdin, stdout, stderr = self.conn.exec_command("whoami /priv")
+                stdout = stdout.read().decode("utf-8", errors="ignore")
+                self.server_os_platform = "Windows"
+                self.user_principal = "admin"
+                if "SeDebugPrivilege" in stdout:
+                    self.admin_privs = True
+                elif "SeUndockPrivilege" in stdout:
+                    self.admin_privs = True
+                    self.user_principal = "admin (UAC)"
                 else:
-                    self.db.add_admin_user(
-                        "plaintext",
-                        username,
-                        password,
-                        host_id=host_id,
-                        cred_id=cred_id,
-                    )
+                    # non admin (low priv)
+                    self.user_principal = "admin (low priv)"
+
+            if not stdout:
+                self.logger.debug(f"User: {self.username} can't get a basic shell")
+                self.server_os_platform = "Network Devices"
+                shell_access = False
             else:
-                stdin, stdout, stderr = self.conn.exec_command("id")
-                output = stdout.read().decode("utf-8")
-                if not output:
-                    self.logger.debug(f"User cannot get a shell")
-                    shell_access = False
-                else:
-                    shell_access = True
+                shell_access = True
+
+            if shell_access and self.server_os_platform == "Linux":
+                self.check_if_admin()
+                if self.admin_privs:
+                    self.logger.debug(f"User {username} logged in successfully and is root!")
+                    if self.args.key_file:
+                        self.db.add_admin_user("key", username, password, host_id=host_id, cred_id=cred_id)
+                    else:
+                        self.db.add_admin_user(
+                            "plaintext",
+                            username,
+                            password,
+                            host_id=host_id,
+                            cred_id=cred_id,
+                        )
 
             self.db.add_loggedin_relation(cred_id, host_id, shell=shell_access)
 
             if self.args.key_file:
-                password = f"{password} (keyfile: {self.args.key_file})"
+                password = f"(keyfile: {self.args.key_file})"
```

Comment:
Why removing the password from the shell output?

Reply:
I think ‘{password}’ is useless here, cause we use private key file

Reply:
Solved!